dcicutils/ff_utils.py

```diff
@@ -941,6 +942,40 @@ def _get_es_metadata(uuids, es_client, filters, sources, chunk_size, auth):
                 yield hit['_source']  # yield individual items from ES
 
 
+def get_schemas(key=None, ff_env=None) -> Dict[str, Dict]:
+    """
+    Gets a dictionary of all schema definitions
+
+    Args:
+        key (dict):   standard ff_utils authentication key
+        ff_env (str): standard ff environment string
+
+    Returns:
+        dict: a mapping from keys that are schema names to schema definitions
+    """
+    auth = get_authentication_with_server(key, ff_env)
+    schemas = get_metadata('profiles/', key=auth, add_on='frame=raw')
+    return schemas
+
+
+def get_schema(name, key=None, ff_env=None) -> Dict:
+    """
+    Gets the schema definition with the given name.
+
+    Args:
+        name (str):   a schema name (CamelCase or snake_case), or None
+        key (dict):   standard ff_utils authentication key
+        ff_env (str): standard ff environment string
+
+    Returns:
+        dict: contains key schema names and value item class names
+    """
+    auth = get_authentication_with_server(key, ff_env)
+    url = f"profiles/{to_camel_case(name)}.json"
+    schema = get_metadata(url, key=auth, add_on='frame=raw')
+    return schema
+
+
 def get_schema_names(key=None, ff_env=None):
```

Comment:
Does it not make sense to call get_schemas in this function to remove redundant code?

Reply:
Thanks for weighing in, @aschroed. It looks like `get_schema_names` does some extra filtering so isn't quite redundant, but I'm thinking _probably_ we don't need them to be disagreeing,. we just need to figure out which we want. My sense is that `get_schemas` should do that same filtering? That would make `get_schema_names` stay full compatible but with less code, as you suggest. (There's _some_ reason to ask why the endpoint doesn't do that filtering. but I'll count that as out-of-scope for now.) My intuition is that I should steal the additional filtering that `get_schema_names` does and have `get_schemas` only return the schemas that `get_schema_names` presently would name. Do you agree?

Reply:
@willronchetti and @drio18, feel free to venture an opinion.

Reply:
I would suggest not including the filtering in `get_schemas`, leaving it as is to have a broad, generic function; the retrieval of schemas plus the filtering could always be a separate function. I interpreted Andy's comment as refactoring `get_schema_names` to call  `get_schemas(key, ff_env)` rather than refactoring `get_schemas` to perform the filtering in `get_schema_names`.

Reply:
The extra filtering occurs after the redundant bit so I was just suggesting replacing 991-993 to a single call to get_schemas.  However, while investigating this I didn't find this function used at all and in fact I found a function of the same name that essentially does the same thing in the functions used in dcicwrangling notebooks - however, that function is now broken because of the fourfront schema upgrade.  And that function does not have any of the filtering. 
There is also a schema object in Submit4DN that does a similar thing but does not depend on the $id property so is not broken and not sure if it is worth changing that but the upshot is I need to do some refactoring of dcicwrangling and I will use these functions here and remove the redundant broken code there.   

I think it depends on the use case whether or not Abstract type schemas are included or not in the get_schemas function, I don't have a strong opinion but perhaps leaving filtering to calling functions makes most sense?

Reply:
Oh - and I have no idea about the local schemas without the id or $id property - maybe for testing only and should be updated to reflect actual schemas?

Reply:
I parameterized it. Can you look at what I did and see if it looks OK?

Reply:
Looks good to me.